src/panels/config/entity_registry/dialog-entity-registry-detail.ts

```diff
@@ -0,0 +1,191 @@
+    await this.updateComplete;
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this._params) {
+      return html``;
+    }
+    const entry = this._params.entry;
+    const stateObj: HassEntity | undefined = this.hass.states[entry.entity_id];
+    const invalidDomainUpdate =
+      computeDomain(this._entityId.trim()) !==
+      computeDomain(this._params.entry.entity_id);
+
+    return html`
+      <paper-dialog
+        with-backdrop
+        opened
+        @opened-changed="${this._openedChanged}"
+      >
+        <h2>${entry.entity_id}</h2>
+        <paper-dialog-scrollable>
+          ${!stateObj
+            ? html`
+                <div>This entity is not currently available.</div>
+              `
+            : ""}
+          ${this._error
+            ? html`
+                <div class="error">${this._error}</div>
+              `
+            : ""}
+          <div class="form">
+            <paper-input
+              .value=${this._name}
+              @value-changed=${this._nameChanged}
+              .label=${this.hass.localize("ui.dialogs.more_info_settings.name")}
+              .placeholder=${stateObj ? computeStateName(stateObj) : ""}
+              .disabled=${this._submitting}
+            ></paper-input>
+            <paper-input
+              .value=${this._entityId}
+              @value-changed=${this._entityIdChanged}
+              .label=${this.hass.localize(
+                "ui.dialogs.more_info_settings.entity_id"
+              )}
+              error-message="Domain needs to stay the same"
+              .invalid=${invalidDomainUpdate}
+              .disabled=${this._submitting}
+            ></paper-input>
+          </div>
+        </paper-dialog-scrollable>
+        <div class="paper-dialog-buttons">
+          <paper-button
+            class="danger"
+            @click="${this._deleteEntry}"
+            .disabled=${this._submitting}
+          >
+            DELETE
+          </paper-button>
+          <paper-button
+            @click="${this._updateEntry}"
+            .disabled=${invalidDomainUpdate || this._submitting}
+          >
+            UPDATE
+          </paper-button>
+        </div>
+      </paper-dialog>
+    `;
+  }
+
+  private _nameChanged(ev: PolymerChangedEvent<string>): void {
+    this._error = undefined;
+    this._name = ev.detail.value;
+  }
+
+  private _entityIdChanged(ev: PolymerChangedEvent<string>): void {
+    this._error = undefined;
+    this._entityId = ev.detail.value;
+  }
+
+  private async _updateEntry(): Promise<void> {
+    try {
+      this._submitting = true;
+      await this._params!.updateEntry({
+        name: this._name.trim() || null,
+        new_entity_id: this._entityId.trim(),
+      });
+      this._params = undefined;
+    } catch (err) {
+      this._submitting = false;
+      this._error = err;
+    }
+  }
+
+  private async _deleteEntry(): Promise<void> {
+    this._submitting = true;
```

Comment:
Should set to false when done.

Reply:
https://github.com/home-assistant/home-assistant-polymer/pull/2643